### PR TITLE
Add the eksd tests job

### PR DESCRIPTION
**Pull request by hellozee**

### CODE REVIEW

File: tetrateci/create_eksd_cluster.sh [hunk 1]
```diff
@@ -20,26 +20,25 @@ SUFFIX=$(sed 's/\.//g' <<< $VER)
 ## Cluster name has to end with k8s.local
 CLUSTER_NAME="test-istio-$SHA8-$SUFFIX.k8s.local"
 
-## TODO: Change to appropriate directory
+CURRENT_DIR=$(pwd)
+
 git clone https://github.com/aws/eks-distro.git
```
Comment (deva26) [inline tetrateci/create_eksd_cluster.sh:25]: It is better to clone this outside of the current repo

File: tetrateci/create_eksd_cluster.sh [hunk 2]
```diff
@@ -20,26 +20,25 @@ SUFFIX=$(sed 's/\.//g' <<< $VER)
 ## Cluster name has to end with k8s.local
 CLUSTER_NAME="test-istio-$SHA8-$SUFFIX.k8s.local"
 
-## TODO: Change to appropriate directory
+CURRENT_DIR=$(pwd)
```
Comment (deva26) [inline tetrateci/create_eksd_cluster.sh:23]: lets do everything relative to basedir which is the repo root.

File: tetrateci/create_eksd_cluster.sh [hunk 3]
```diff
@@ -20,26 +20,25 @@ SUFFIX=$(sed 's/\.//g' <<< $VER)
...
-
+# possible versions: 1-18, 1-19
+export RELEASE_BRANCH=${VER}
 
 echo "creating a eksd cluster with \"$CLUSTER_NAME\" name..."
 ./run_cluster.sh
 
 #Wait for the cluster to be created
-#TODO - wait of 6minutes may not be enough. Loop this call twice.
 ./cluster_wait.sh
```
Comment (deva26) [inline tetrateci/create_eksd_cluster.sh:42]: Is the time in cluster_wait of 6 mins enough for this?

Comment (hellozee) [reply tetrateci/create_eksd_cluster.sh:44]: was enough in my tests

File: .github/workflows/e2e_tests.yml [hunk 4]
```diff
@@ -253,13 +253,78 @@ jobs:
...
+          path: ~/go/pkg/mod
+          key: ${{ steps.get_minor_ver.outputs.REL_BRANCH_VER }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ steps.get_minor_ver.outputs.REL_BRANCH_VER }}-go-
+
+      - name: eksd-cluster-create
+        run: bash ./tetrateci/create_eksd_cluster.sh
+        env:
+          VER: ${{ matrix.version }}
+          MINOR_VER: ${{ steps.get_minor_ver.outputs.REL_BRANCH_VER }}
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:301]: Can you bring the new naming used (ISTIO_MINOR_VER and K8S_VERSION) here?

Comment (hellozee) [reply .github/workflows/e2e_tests.yml:301]: Yes, but first I would need to rebase onto the k8s patch, cause I am afraid there would be a bunch of merge conflicts.

File: tetrateci/create_eksd_cluster.sh [hunk 5]
```diff
@@ -20,26 +22,25 @@ SUFFIX=$(sed 's/\.//g' <<< $K8S_VERSION)
...
+export RELEASE_BRANCH=$(sed 's/\./-/g' <<< $K8S_VERSION)
 
 echo "creating a eksd cluster with \"$CLUSTER_NAME\" name..."
 ./run_cluster.sh
 
 #Wait for the cluster to be created
-#TODO - wait of 6minutes may not be enough. Loop this call twice.
 ./cluster_wait.sh
 
+cd $CURRENT_DIR
```
[the remark this answers is not in the document]
Comment (hellozee) [reply tetrateci/create_eksd_cluster.sh:46]: oh wait, I did change this, but probably got lost in the merge

File: tetrateci/test_1.9.sh [hunk 6]
```diff
@@ -8,6 +8,8 @@ echo "Applying patches...."
 git apply tetrateci/patches/common/disable-dashboard.1.9.patch
 git apply tetrateci/patches/common/disable-ratelimiting.1.9.patch
 git apply tetrateci/patches/common/increase-vm-timeout.1.9.patch
+git apply tetrateci/patches/common/increase-sniffing-timeout.1.9.patch
```
Comment (deva26) [inline tetrateci/test_1.9.sh:11]: Address the patch fixes in another PR. Remove these changes from here.

**Review by deva26 — APPROVED** [review]

**Pull request merged**